version: '3.3'

services:

  rabbitmq:
    image: sammlerio/rabbitmq:0.1.3
    container_name: rabbitmq
    # Seems to throw an error on Mac-OS
#    volumes:
#      - ./.data/rabbitmq/data:/var/lib/rabbitmq
#      - ./.data/rabbitmq/log:/var/log/rabbitmq/sammler
    ports:
      - "15672:15672"
      - "5672:5672"
    environment:
      - RABBITMQ_NODENAME=sammler
      - DOCKER_LABELS=true
      - LOGSPOUT=ignore
    labels:
      - io.sammler.env=DEV
      - io.sammler.service=rabbitmq
    depends_on:
      - logstash
    restart: always

  log-service:
    image: sammlerio/log-service:0.0.5@sha256:c37384045355bef9b46ec132c3ea771e5c4964d85967102fcd37e7a26243b3eb
    container_name: log-service
    environment:
      - NOVE_ENV=production
      - SERVICE_NAME=log-service
      - MONG

  strategy-heartbeat:
    image: sammlerio/strategy-heartbeat:0.1.0@sha256:5c10db17c18b497c81e2ff8f42f67a0f2df2328011026b137e00f7c07ce346f1
    container_name: strategy-heartbeat
    environment:
      - NODE_ENV=production
      - SERVICE_NAME=strategy-heartbeat
      - RABBITMQ_URI=amqp://guest:guest@s5r-rabbitmq:5672
      - LOG_SERVICE_URI=http://log-service:3004
    volumes:
      - ./src:/opt/strategy-heartbeat
